src/amo/components/CollectionManager/index.js

```diff
@@ -88,16 +99,30 @@ export class CollectionManagerBase extends React.Component<Props, State> {
       return;
     }
 
-    dispatch(updateCollection({
-      collectionSlug: collection.slug,
-      defaultLocale: collection.defaultLocale,
+    if (!this.state.slug) {
+      errorHandler.addMessage(
+        i18n.gettext('Collection slug cannot be blank')
+      );
+      return;
+    }
+
+    const payload: Object = {
+      collectionSlug: (collection && collection.slug) || undefined,
+      defaultLocale: (collection && collection.defaultLocale) || siteLang,
       description: { [siteLang]: this.state.description },
       errorHandlerId: errorHandler.id,
-      formOverlayId: COLLECTION_OVERLAY,
+      formOverlayId:
+        creating ? CREATE_COLLECTION_OVERLAY : EDIT_COLLECTION_OVERLAY,
       name: { [siteLang]: this.state.name },
-      user: collection.authorUsername,
+      user: (collection && collection.authorUsername) || currentUsername,
       slug: this.state.slug,
-    }));
+    };
+
+    if (creating) {
+      dispatch(createCollection(payload));
```

Comment:
FTR, it was these calls that Flow was upset about when I tried to use a more specific type above. And adding `invariant` into the different cases for different mandatory parameters in here didn't help.